src/HealthGPS/analysis_module.cpp

```diff
@@ -26,6 +26,41 @@ SimulationModuleType AnalysisModule::type() const noexcept {
     return SimulationModuleType::Analysis;
 }
 
+void AnalysisModule::initialise_vector(RuntimeContext &context) {
+    // Set some arbitrary factors
+    std::vector<core::Identifier> factors{"Gender"_id, "Age"_id};
+
+    std::vector<int> factor_bins;
+
+    for (const auto &factor : factors) {
+        auto min_max = std::minmax_element(
+            context.population().cbegin(), context.population().cend(),
+            [&factor](const auto &a, const auto &b) {
+                return a.get_risk_factor_value(factor) < b.get_risk_factor_value(factor);
+            });
```

Comment:
Use the [std::ranges version](https://en.cppreference.com/w/cpp/algorithm/ranges/minmax_element):

`std::minmax_element(context.population(), ...`